### PR TITLE
Ea 2364 entity retrieval

### PR DESCRIPTION
**Pull request by SrdjanStevanetic**
Hi Sergiu, please review the changes. I have added the xml serializer.

### CODE REVIEW

**Comment by SrdjanStevanetic** [comment]
@gsergiu Please review the changes.

File: entity-management-definitions/src/main/java/eu/europeana/entitymanagement/definitions/model/Agent.java [hunk 1]
```diff
@@ -4,7 +4,7 @@
 import java.util.List;
 import java.util.Map;
 
-public interface Agent extends Entity {
+public interface Agent extends EntityRoot {
```
Comment (gsergiu) [inline entity-management-definitions/src/main/java/eu/europeana/entitymanagement/definitions/model/Agent.java]: Is it really needed to change the EnityInterface to EntityRoot?

File: entity-management-definitions/src/main/java/eu/europeana/entitymanagement/definitions/model/vocabulary/WebEntityFields.java [hunk 2]
```diff
@@ -45,7 +46,7 @@
 	public static final String BROAD_MATCH = "broadMatch";
 	public static final String NARROW_MATCH = "narrowMatch";
 	public static final String RELATED_MATCH = "relatedMatch";
-	public static final String IN_SCHEME = "inScheme";
+	public static final String IN_SCHEMA = "inSchema";
```
Comment (gsergiu) [inline entity-management-definitions/src/main/java/eu/europeana/entitymanagement/definitions/model/vocabulary/WebEntityFields.java]: please reverts the correct name of the field is inScheme

File: entity-management-definitions/src/main/java/eu/europeana/entitymanagement/definitions/model/Agent.java (hunk 1 above)
Comment (gsergiu) [reply entity-management-definitions/src/main/java/eu/europeana/entitymanagement/definitions/model/Agent.java]: Please revert back the renaming.

File: entity-management-mongo/src/main/java/eu/europeana/entitymanagement/mongo/repository/EntityRecordRepository.java [hunk 3]
```diff
@@ -29,7 +30,7 @@
      * @return the total number of resources in the database
      */
     public long count() {
-        return datastore.find(EntityRecordImpl.class).count();
+        return datastore.find(BaseEntityRecord.class).count();
```
Comment (gsergiu) [inline entity-management-mongo/src/main/java/eu/europeana/entitymanagement/mongo/repository/EntityRecordRepository.java:33]: The type of the objects in the database must be the EntityRecordImpl not the BaseEntityRecord

**Review by gsergiu — CHANGES_REQUESTED** [review]
Please perform the required changes

File: entity-management-definitions/src/main/java/eu/europeana/entitymanagement/definitions/model/Agent.java (hunk 1 above)
Comment (SrdjanStevanetic) [reply entity-management-definitions/src/main/java/eu/europeana/entitymanagement/definitions/model/Agent.java]: Done.

File: entity-management-definitions/src/main/java/eu/europeana/entitymanagement/definitions/model/vocabulary/WebEntityFields.java (hunk 2 above)
Comment (SrdjanStevanetic) [reply entity-management-definitions/src/main/java/eu/europeana/entitymanagement/definitions/model/vocabulary/WebEntityFields.java]: Done.

File: entity-management-mongo/src/main/java/eu/europeana/entitymanagement/mongo/repository/EntityRecordRepository.java (hunk 3 above)
Comment (SrdjanStevanetic) [reply entity-management-mongo/src/main/java/eu/europeana/entitymanagement/mongo/repository/EntityRecordRepository.java:33]: We have discussed to keep this so far as it is.

**Pull request merged**